contract/src/msgs/data_requests/sudo/remove_requests.rs

```diff
@@ -43,67 +43,78 @@ fn remove_request(
     let mut dr_escrow = DR_ESCROW.load(deps.storage, &dr_id)?;
 
     // add 1 so we can account for the refund message that may be sent
-    let mut bank_messages = Vec::with_capacity(messages.messages.len() + 1);
-    for message in messages.messages {
-        match &message.kind {
-            DistributionKind::Burn(distribution_burn) => {
-                dr_escrow.amount = dr_escrow.amount.checked_sub(distribution_burn.amount)?;
+    let mut bank_messages = Vec::new();
+
+    // We need to send messages in the order given, which should be: burn -> data_proxy_reward -> executor_reward
+    for message in messages {
+        // No reason to keep processing if the escrowed amount is zero
+        if dr_escrow.amount.is_zero() {
+            break;
+        }
```

Comment:
We should emit something here, because this should never happen.

Reply:
Added a message but not sure it should say. I thought this could happen/thought this was an intended feature.